src/core/federation.ts

```diff
@@ -839,17 +840,16 @@ export class FederatedApp extends App {
         let currentTime = this.util.getCurrentLogicalTime()
         if (currentTime.isEarlierThan(nextTag.time)) {
             // Tell the RTI logical time is being advanced to a greater value.
-            this.sendRTILogicalTimeComplete(currentTime);
+            this.sendRTILogicalTimeComplete(nextTag.time);
```

Comment:
Check c implementation and see when the LTC (Logical Time Complete) should be sent. (Probably it should be sent after processing everything in the event queue.) And make the naming make sense.

Reply:
See what MSG_TYPE_LOGICAL_TAG_COMPLETE message is for.

https://github.com/lf-lang/reactor-c/blob/main/core/federated/net_common.h#L403

Reply:
In reality, LTC and NET can be combined into an NMR (Next Message Request) message.
LTC is supposed to indicate to the RTI that a federate has finished processing a particular tag, usually followed by a NET that is indicating to the RTI what tag it is going to execute next, absent any unforeseen incoming network messages.

Reply:
Our reference implementation of the RTI in C expects separate LTC and NET messages, but the comment above was meant to give some context about LTC and its relation to NET.

Reply:
Thanks for the explanation, @Soroosh129! Is this protocol described anywhere but in the comments of the C implementation? In conclusion, the used terminology makes sense, but the implementation has been wrong all along. The argument should be `currentTime` like it was, but the statement should be in a different place altogether, in a hook that executes once all reactions have executed and the reaction queue is empty.

Reply:
Yes, this is explained in an (the) upcoming paper.

Some details about the federated execution is also explained in the "Quantifying the CAP Theorem" paper.